File: algorithms/2_medium/R/02091_removing_minimum_and_maximum_from_array/solution.c
int minimumDeletions(const int *restrict nums, const int len) {
    if ((len > 0 && nums == NULL) || len < 0) { return -1; }
    if (len <= 2) { return len; }

    int max = nums[0];
    int max_idx = 0;
    int min = nums[0];
    int min_idx = 0;

    for (int i = 1; i < len; i += 1) {
        const int num = nums[i];

        if (num > max) {
            max = num;
            max_idx = i;
        }

        if (num < min) {
            min = num;
            min_idx = i;
        }
    }

    const int r = MAX(max_idx, min_idx);
    const int l = MIN(max_idx, min_idx);

    const int from_r = len - r;
    const int from_l = l + 1;
    const int from_mid = r - l;

    return MIN(
        from_l + from_r,
        MIN(from_l + from_mid, from_mid + from_r)
    );
}

File: algorithms/2_medium/R/02091_removing_minimum_and_maximum_from_array/solution.rs
impl Solution {
    pub fn minimum_deletions(nums: Vec<i32>) -> i32 {
        if nums.len() <= 2 { return nums.len() as i32; }

        let len = nums.len();
        let (mut max, mut max_idx) = (nums[0], 0_usize);
        let (mut min, mut min_idx) = (nums[0], 0_usize);

        for i in 1..len {
            let num = nums[i];

            if num > max {
                max = num;
                max_idx = i;
            }

            if num < min {
                min = num;
                min_idx = i;
            }
        }

        let r = max_idx.max(min_idx);
        let l = max_idx.min(min_idx);

        let from_r = len - r;
        let from_l = l + 1;
        let from_mid = r - l;

        (from_l + from_r).min(from_l + from_mid).min(from_mid + from_r).try_into().unwrap()
    }
}
